crates/primitives/starknet/src/tests/utils.rs

```diff
@@ -66,5 +74,11 @@ pub fn create_test_state() -> CachedState<DictStateReader> {
 pub fn get_contract_class(contract_path: &str, version: u8) -> ContractClass {
     let path: PathBuf = [contract_path].iter().collect();
     let raw_contract_class = fs::read_to_string(path).unwrap();
-    starknet_serde::get_contract_class(&raw_contract_class, version)
```

Comment:
`read_contract_class_from_json` is defined in the pallet_starknet crate. I don't want to have primitive depend on it just so it can read a file from the disk for its tests. And it would probably create a circular dependency situation.

Code duplication of a helper function is not a huge problem. It prefers it than the other solution I could though of

Reply:
`get_contract_class ` was defined inside primitives earlier. So if we define `read_contract_class_from_json` inside primitives

- It will be accessible by the pallet as the pallet is already importing the primitives
- And we can use it in this test case as well

Do we not want to have it in primitives?

Reply:
It's not really a `primitive` to anything. It's a simple helper function that reads from the disk.
More like a `utils`.

It exists two times, once behind a config flag, and once for the tests. I have no problem copy-pasting it in the test.
If it appears a third time somewhere, why not share it again, for now I prefer to have it closer to the only place it is actually used (outside of tests).

Basically, I don't want to modify the release code for the testing one. It is only used once in the executed codebase, so let's keep it there. I don't mind duplication once for a test

Reply:
Oh ok, I get your point, makes sense then